### PR TITLE
Reactivating macOS workflow

### LINKED ISSUE #146: Support the build of macOS
### 📝 Description of the feature

From version ``0.51.8`` onwards (https://github.com/pyansys/pymapdl-reader/pull/93) support for macOS builds was removed. Package is not available in pip from ``0.51.7`` onwards. Following the PyAnsys metapackage creation (and ``ansys-mapdl-reader`` being an optional dependency of it), we should try and support it.

### 💡 Steps for implementing the feature

Try reactivating and updating current workflow to take into account macOS as well

### 🔗 Useful links and references

_No response_

### PR DESCRIPTION
**Pull request by RobPasMue**
Resolves #146

### CODE REVIEW

**Comment by RobPasMue** [comment]
PR ready for review @akaszynski @germa89 - we would need to create a release for this so that the wheels are uploaded and solve the issues mentioned on #146

**Comment by akaszynski** [comment]
@RobPasMue, recommend we use `cibuildwheel`. See https://github.com/pyvista/pymeshfix/pull/44

File: .github/workflows/testing-and-deployment.yml [hunk 1]
```diff
@@ -151,7 +147,7 @@ jobs:
           pip install twine
           twine check dist/*
 
-      - name: Install on Linux
+      - name: Install on Linux / macOS
```
Comment (akaszynski) [inline .github/workflows/testing-and-deployment.yml:150]:
```suggestion
      - name: Install on Linux
```

**Review by akaszynski — APPROVED** [review]
image

LGTM.

**Comment by RobPasMue** [comment]
Thanks @akaszynski for taking care of this!

**Pull request merged**